### PR TITLE
Annotate Windows Server 2012 as deprecated

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
This PR adds an annotation to the Windows Server 2012 R2 template to tag it as deprecated.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Windows Server 2012 R2 is now deprecated
```

### CODE REVIEW

File: automation/test-windows.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 set -ex
 
 namespace="kubevirt"
-template_name="windows2k12r2"
+template_name=""
```
Comment (0xFelix) [inline automation/test-windows.sh:6]: This will break the script if no `TARGET` env was provided.

Comment (jean-edouard) [reply automation/test-windows.sh:6]: Right, thanks for the review, do you have a preferred default?

Comment (0xFelix) [reply automation/test-windows.sh:6]: Let's go with windows2k22 as the current successor.

Please note that this also needs adjustment of the CI jobs in openshift/release. At least the 2k12 lane needs to be removed.

Comment (jean-edouard) [reply automation/test-windows.sh]: Default fixed, and created https://github.com/openshift/release/pull/42370

**Comment by lyarwood** [comment]
I'm against this, looking at the docs:

https://learn.microsoft.com/en-us/lifecycle/announcements/windows-server-2012-r2-end-of-support

> Windows Server 2012 and Windows Server 2012 R2 will end on October 10, 2023. After this date, these products will no longer receive security updates, non-security updates, bug fixes, technical support, or online technical content updates. If you cannot upgrade to the next version, you will need to use Extended Security Updates (ESUs) for up to three years. ESUs are available for free in Azure or need to be purchased for on-premises deployments. Microsoft has guidance for both cloud and on-premises solutions. Go [here](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/three-options-to-prepare-for-windows-server-2012-r2-end-of/ba-p/3645211) to learn more.

So it's still available through EUS, would it be better to mark the templates as deprecated and drop CI coverage for now?

**Comment by jean-edouard** [comment]
> I'm against this, looking at the docs:
> 
> https://learn.microsoft.com/en-us/lifecycle/announcements/windows-server-2012-r2-end-of-support
> 
> > Windows Server 2012 and Windows Server 2012 R2 will end on October 10, 2023. After this date, these products will no longer receive security updates, non-security updates, bug fixes, technical support, or online technical content updates. If you cannot upgrade to the next version, you will need to use Extended Security Updates (ESUs) for up to three years. ESUs are available for free in Azure or need to be purchased for on-premises deployments. Microsoft has guidance for both cloud and on-premises solutions. Go [here](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/three-options-to-prepare-for-windows-server-2012-r2-end-of/ba-p/3645211) to learn more.
> 
> So it's still available through EUS, would it be better to mark the templates as deprecated and drop CI coverage for now?
> 
> That's going to be my approach in common-instancetypes anyway:
> 
> [kubevirt/common-instancetypes#99](https://github.com/kubevirt/common-instancetypes/pull/99)

Done, PTAL!

**Comment by lyarwood** [comment]
/approve

Thanks! If you respin please include the link to the support docs in the commit and description of the PR for context.

**Comment by lyarwood** [comment]
/lgtm

Thanks! If you respin please include the link to the support docs in the commit and description of the PR for context.

**Pull request merged**